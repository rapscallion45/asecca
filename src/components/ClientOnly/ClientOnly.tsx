import { FC, useState, useEffect, ReactNode } from 'react';

interface IClientOnlyProps {
  children: ReactNode;
}

/**
 * Client Only - renders component on client only, no SSR
 *
 * @author - [Carl Scrivener](https://github.com/rapscallion45)
 * @since - 0.0.0
 *
 * @param props - component children
 * @returns {FC} - client only HOC
 */
const ClientOnly: FC<IClientOnlyProps> = (props) => {
  const { children } = props;
  const [hasMounted, setHasMounted] = useState<boolean>(false);

  /** useEffect is only run in the client/browser environment */
  useEffect(() => {
    /** useEffect has run, therefore this is browser/client environment */
    setHasMounted(true);
  }, []);

  return hasMounted ? (
    <div aria-labelledby="client-only-element">{children}</div>
  ) : null;
};

export default ClientOnly;
