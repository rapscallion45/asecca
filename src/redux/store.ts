import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import logger from 'redux-logger';
import userPermissionReducer from './slices/userPermissionSlice';
import costsConfigReducer from './slices/costsConfigSlice';
import notificationsReducer from './slices/notificationsSlice';
import themeReducer from './slices/themeSlice';
import { collectionsKanbanSlice } from './slices/kanbanSlice';

/**
 * Configuration of the redux store to manage the global application state
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on Redux Store object](https://redux.js.org/api/store)
 *
 * @constant
 * @type {ToolkitStore<Object>}
 */
const store = configureStore({
  reducer: {
    userPermission: userPermissionReducer,
    costsConfig: costsConfigReducer,
    notifications: notificationsReducer,
    theme: themeReducer,
    collectionsKanban: collectionsKanbanSlice.reducer,
  },
  // @ts-ignore
  middleware:
    process.env.NODE_ENV === 'development'
      ? (getDefaultMiddleware) => getDefaultMiddleware().concat(logger)
      : (getDefaultMiddleware) => getDefaultMiddleware(),
});

/**
 * Shorthand application store dispatch type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppDispatch - application specific store dispatch type
 */
export type AppDispatch = typeof store.dispatch;

/**
 * Shorthand application state type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppState - application specific state object shape
 */
export type AppState = ReturnType<typeof store.getState>;

/**
 * Shorthand application Thunk Action type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppThunk - application specific Thunk action definition
 */
export type AppThunk = ThunkAction<void, AppState, unknown, Action<string>>;

export default store;
