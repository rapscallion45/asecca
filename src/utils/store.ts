/* redux store configuration placeholder */
const useStore = () => {};
export default useStore;
