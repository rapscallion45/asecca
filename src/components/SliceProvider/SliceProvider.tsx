import { FC, ReactNode, createContext, useContext } from 'react';
import { useSelector } from 'react-redux';
import { Slice } from '@reduxjs/toolkit';
import { AppState } from '@/redux/store';

/**
 * Default value for the slice context
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.8
 * @memberof SliceProvider
 *
 * @constant
 * @type {Slice<Object>}
 */
const SliceContext = createContext<Slice>({} as Slice);

/**
 * Slice Provider Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.8
 *
 * @typedef ISliceProviderProps
 * @prop {Slice} slice - context slice to provide down to child nodes
 * @prop {ReactNode} children - child nodes
 */
interface ISliceProviderProps {
  slice: Slice;
  children: ReactNode;
}

/**
 * Slice Provider
 *
 * Slice provider functional component wrapper for providing global state
 * managment functionality to child components
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.8
 *
 * @component
 * @param {ISliceProviderProps} props - component props
 * @returns {FC} - slice provider functional component
 */
const SliceProvider: FC<ISliceProviderProps> = (props) => {
  const { slice, children } = props;
  return (
    <SliceContext.Provider value={slice}>{children}</SliceContext.Provider>
  );
};

/**
 * Provider specific actions interface for passed slice
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.8
 * @memberof KanbanReduxSlice
 *
 * @see See [more info on Redux Slice creation](https://redux-toolkit.js.org/api/createSlice)
 *
 * @constant
 * @type {CaseReducerActions<any, string>}
 */
export const useSliceActions = () => useContext<Slice>(SliceContext).actions;

/**
 * Provider specific selector interface for passed slice
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.8
 * @memberof KanbanReduxSlice
 *
 * @see See [more info on Redux Slice creation](https://redux-toolkit.js.org/api/createSlice)
 *
 * @constant
 * @type {AppState}
 */
export const useSliceSelector = () => {
  const { name } = useContext<Slice>(SliceContext);
  return useSelector((state: AppState) => state[name as keyof AppState]);
};

export default SliceProvider;
