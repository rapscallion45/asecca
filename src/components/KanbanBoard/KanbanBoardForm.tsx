import { FC, useState, useCallback } from 'react';
import {
  Box,
  IconButton,
  Button,
  TextField,
  FormControl,
  Input,
  InputAdornment,
  CircularProgress,
  Typography,
} from '@mui/material';
import CloseIcon from '@mui/icons-material/Close';
import LabelImportantIcon from '@mui/icons-material/LabelImportant';
import { IKanbanBoardColumn } from '@/lib/api/api-types';
import { IKanbanBoardState } from '@/redux/types';
import useKanbanBoardFormController from './KanbanBoardFormController';
import { useSliceSelector } from '../SliceProvider/SliceProvider';

/**
 * Kanban Board Form Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.3
 *
 * @typedef IKanbanBoardFormProps
 * @prop {boolean} isEditMode - determines whether this is a new board or editing
 * @prop {boolean} canEdit - board data can be edited
 * @prop {any} closeModal - on close modal callback handler
 */
interface IKanbanBoardFormProps {
  isEditMode: boolean;
  canEdit?: boolean;
  closeModal?: () => void;
}

/**
 * Kanban Board Form
 *
 * Kanban board form interface for collecting user input when creating
 * or editing a board's details
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.3
 *
 * @component
 * @param {IKanbanBoardFormProps} props - component props
 * @returns {FC} - kanaban board form functional component
 */
const KanbanBoardForm: FC<IKanbanBoardFormProps> = (props) => {
  const { isEditMode, canEdit = false, closeModal } = props;
  const { data: currentData } = useSliceSelector() as IKanbanBoardState;
  const [newColumns, setNewColumns] = useState<Array<IKanbanBoardColumn>>(
    currentData?.columns || [
      { name: 'Todo', tasks: [] },
      { name: 'In Progress', tasks: [] },
      { name: 'Completed', tasks: [] },
    ]
  );
  const { saving, formik } = useKanbanBoardFormController(
    isEditMode,
    newColumns,
    closeModal
  );

  /**
   * Callback handler for user input updates to columns
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.3
   *
   * @method
   * @param {string} colIdx - column index that has been updated
   * @param {string} newValue - updated value for column name
   */
  const onChange = useCallback((colIdx: number, newValue: string) => {
    setNewColumns((prevState: Array<IKanbanBoardColumn>) => {
      const newState = [...prevState];
      const column = newState.find(
        (col: IKanbanBoardColumn, index: number) => index === colIdx
      );
      if (column) column.name = newValue;
      return newState;
    });
  }, []);

  /**
   * Callback handler for column deletion
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.3
   *
   * @method
   * @param {number} id - column ID to be deleted
   */
  const onDelete = useCallback((colIdx: number) => {
    setNewColumns((prevState) =>
      prevState.filter(
        (el: IKanbanBoardColumn, index: number) => index !== colIdx
      )
    );
  }, []);

  return (
    <form onSubmit={formik.handleSubmit}>
      <TextField
        fullWidth
        variant="outlined"
        margin="normal"
        id="name"
        name="name"
        label="Board Name"
        type="text"
        disabled={!canEdit}
        value={formik.values.name}
        onChange={formik.handleChange}
        error={formik.touched.name && Boolean(formik.errors.name)}
        helperText={formik.touched.name && formik.errors.name}
        autoComplete="on"
        InputProps={{
          startAdornment: (
            <InputAdornment position="start">
              <LabelImportantIcon color="secondary" />
            </InputAdornment>
          ),
        }}
      />
      <Box mt={2}>
        <Typography>Board Columns</Typography>
        {newColumns.map((column: IKanbanBoardColumn, index: number) => (
          <Box key={column.name}>
            <FormControl fullWidth sx={{ m: 1 }} variant="standard">
              <Input
                id="board-columns-form"
                type="text"
                value={column.name}
                onChange={(e) => {
                  onChange(index, e.target.value);
                }}
                disabled={!canEdit}
                placeholder="Enter column name..."
                endAdornment={
                  <InputAdornment position="end">
                    {canEdit && (
                      <IconButton
                        aria-label="delete column"
                        data-testid={`delete-column-${column.name}`}
                        onClick={() => {
                          onDelete(index);
                        }}
                      >
                        <CloseIcon />
                      </IconButton>
                    )}
                  </InputAdornment>
                }
              />
            </FormControl>
          </Box>
        ))}
        <Box mt={2}>
          <Button
            variant="outlined"
            color="secondary"
            onClick={() => {
              setNewColumns((state: Array<IKanbanBoardColumn>) => [
                ...state,
                { name: '', tasks: [], groups: [] },
              ]);
            }}
            fullWidth
            disabled={!canEdit}
          >
            + Add New Column
          </Button>
        </Box>
      </Box>
      {!isEditMode ? (
        <Button
          type="submit"
          fullWidth
          variant="contained"
          color="secondary"
          disabled={saving || !canEdit}
          sx={{ padding: '10px 0', marginTop: '20px' }}
        >
          {!saving && 'Add Board'}
          {saving && <CircularProgress size={25} color="inherit" />}
        </Button>
      ) : (
        <Button
          type="submit"
          fullWidth
          variant="contained"
          color="secondary"
          disabled={saving || !canEdit}
          sx={{ padding: '10px 0', marginTop: '20px' }}
        >
          {!saving && 'Update'}
          {saving && <CircularProgress size={25} color="inherit" />}
        </Button>
      )}
    </form>
  );
};

export default KanbanBoardForm;
