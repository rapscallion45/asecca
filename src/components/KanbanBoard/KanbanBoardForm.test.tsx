import React from 'react';
import { Provider } from 'react-redux';
import {
  render,
  screen,
  waitFor,
  act,
  fireEvent,
} from '@testing-library/react';
import {
  collectionsKanbanSlice,
  fetchByProjectId as fetchKanbanBoardByProjectId,
} from '@/redux/slices/collectionsKanbanSlice';
import store from '../../redux/store';
import KanbanBoardForm from './KanbanBoardForm';
import SliceProvider from '../SliceProvider/SliceProvider';
import { IClientOnlyProps } from '../ClientOnly/ClientOnly';
// import collectionsKanbanDataMock from '../../../__mocks__/Kanban/collectionsKanbanDataMock';
// import { IKanbanBoardColumn } from '@/lib/api/api-types';

/* default test query ID */
const query: string = '66135000001760012';

/* mock close modal callback */
const mockCloseModalCallback = jest.fn(() => {});

/* mock Client Only renderer - just render children */
jest.mock(
  '../ClientOnly/ClientOnly',
  () => (props: IClientOnlyProps) => props.children
);

/**
 * Kanban Board Form Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.22
 */
describe('Kanban Board Form', () => {
  describe('Values', () => {
    /** ensure mock function calls are cleared after test */
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('Should render expected input fields and display values', async () => {
      /** Arrange */
      act(() => {
        store.dispatch(
          fetchKanbanBoardByProjectId({
            projectId: query,
          })
        );
      });

      /** Act */
      render(
        <Provider store={store}>
          <SliceProvider slice={collectionsKanbanSlice}>
            <KanbanBoardForm isEditMode canEdit />
          </SliceProvider>
        </Provider>
      );

      /** Assert - correct value should be displayed from test data */
      // await waitFor(() => {
      //   collectionsKanbanDataMock.collections.columns.forEach(
      //     (column: IKanbanBoardColumn) => {
      //       expect(screen.getByDisplayValue(column.name)).toBeInTheDocument();
      //     }
      //   );
      // });
      expect(screen.getByLabelText('Board Name')).toBeInTheDocument();
      expect(screen.getByText('Board Columns')).toBeInTheDocument();
      expect(screen.getByText('+ Add New Column')).toBeInTheDocument();
      expect(screen.getByText('Update')).toBeInTheDocument();
    });
  });

  describe('Editing and Saving', () => {
    /** ensure mock function calls are cleared after test */
    afterEach(() => {
      jest.clearAllMocks();
    });

    it('Should render button loading spinner when saving data', async () => {
      /** Arrange */
      act(() => {
        store.dispatch(
          fetchKanbanBoardByProjectId({
            projectId: query,
          })
        );
      });

      /** Act */
      render(
        <Provider store={store}>
          <SliceProvider slice={collectionsKanbanSlice}>
            <KanbanBoardForm
              isEditMode
              canEdit
              closeModal={mockCloseModalCallback}
            />
          </SliceProvider>
        </Provider>
      );

      /** Assert - wait for values to be loaded into form */
      // await waitFor(() => {
      //   collectionsKanbanDataMock.collections.columns.forEach(
      //     (column: IKanbanBoardColumn) => {
      //       expect(screen.getByDisplayValue(column.name)).toBeInTheDocument();
      //     }
      //   );
      // });
      expect(screen.getByLabelText('Board Name')).toBeInTheDocument();
      expect(screen.getByText('Board Columns')).toBeInTheDocument();
      expect(screen.getByText('+ Add New Column')).toBeInTheDocument();
      expect(screen.getByText('Update')).toBeInTheDocument();

      /** Act - update values */
      act(() => {
        fireEvent.change(screen.getByLabelText('Board Name'), {
          target: { value: 'Test Board' },
        });
      });
      // act(() => {
      //   fireEvent.change(screen.getByDisplayValue('Todo'), {
      //     target: { value: 'Test Column' },
      //   });
      // });
      // act(() => {
      //   fireEvent(
      //     screen.getByTestId('delete-column-Completed'),
      //     new MouseEvent('click', {
      //       bubbles: true,
      //       cancelable: true,
      //     })
      //   );
      // });

      /** Assert - wait for values to update */
      await waitFor(() => {
        expect(screen.getByDisplayValue('Test Board')).toBeInTheDocument();
      });
      // expect(screen.getByDisplayValue('Test Column')).toBeInTheDocument();
      expect(screen.queryByDisplayValue('Completed')).toBeNull();

      /** Act - click Update button to trigger API save functionality */
      act(() => {
        fireEvent(
          screen.getByText('Update'),
          new MouseEvent('click', {
            bubbles: true,
            cancelable: true,
          })
        );
      });

      /** Assert - Update button clicked and so no longer rendered */
      await waitFor(() => {
        expect(screen.queryByDisplayValue('Update')).toBeNull();
      });
    });
  });

  it('Renders correctly', async () => {
    /** Arrange */
    act(() => {
      store.dispatch(
        fetchKanbanBoardByProjectId({
          projectId: query,
        })
      );
    });

    /** perform snapshot test */
    const tree = render(
      <Provider store={store}>
        <SliceProvider slice={collectionsKanbanSlice}>
          <KanbanBoardForm isEditMode={false} />
        </SliceProvider>
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
});
