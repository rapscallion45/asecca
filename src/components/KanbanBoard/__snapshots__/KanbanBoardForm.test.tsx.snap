// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Kanban Board Form Renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root MuiFormControl-marginNormal MuiFormControl-fullWidth MuiTextField-root css-17vbkzs-MuiFormControl-root-MuiTextField-root"
  >
    <label
      className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-disabled MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink={true}
      htmlFor="name"
      id="name-label"
    >
      Board Name
    </label>
    <div
      className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-disabled MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedStart css-8j6b76-MuiInputBase-root-MuiOutlinedInput-root"
      onClick={[Function]}
    >
      <div
        className="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-ittuaa-MuiInputAdornment-root"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
          data-testid="LabelImportantIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m3.5 18.99 11 .01c.67 0 1.27-.33 1.63-.84L20.5 12l-4.37-6.16c-.36-.51-.96-.84-1.63-.84l-11 .01L8.34 12 3.5 18.99z"
          />
        </svg>
      </div>
      <input
        aria-invalid={false}
        autoComplete="on"
        autoFocus={false}
        className="MuiInputBase-input MuiOutlinedInput-input Mui-disabled MuiInputBase-inputAdornedStart css-1o9s3wi-MuiInputBase-input-MuiOutlinedInput-input"
        disabled={true}
        id="name"
        name="name"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={false}
        type="text"
        value="New Board"
      />
      <fieldset
        aria-hidden={true}
        className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          className="css-14lo706"
        >
          <span>
            Board Name
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
  <div
    className="MuiBox-root css-1yuhvjn"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      Board Columns
    </p>
    <div
      className="MuiBox-root css-0"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-v4u5dn-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={true}
            id="board-columns-form"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter column name..."
            required={false}
            type="text"
            value="Todo"
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-v4u5dn-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={true}
            id="board-columns-form"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter column name..."
            required={false}
            type="text"
            value="In Progress"
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          />
        </div>
      </div>
    </div>
    <div
      className="MuiBox-root css-0"
    >
      <div
        className="MuiFormControl-root MuiFormControl-fullWidth css-1w4vsez-MuiFormControl-root"
      >
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-disabled MuiInputBase-formControl MuiInputBase-adornedEnd css-v4u5dn-MuiInputBase-root-MuiInput-root"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input Mui-disabled MuiInputBase-inputAdornedEnd css-1x51dt5-MuiInputBase-input-MuiInput-input"
            disabled={true}
            id="board-columns-form"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Enter column name..."
            required={false}
            type="text"
            value="Completed"
          />
          <div
            className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-standard MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
          />
        </div>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-fullWidth css-1p5sqbq-MuiButtonBase-root-MuiButton-root"
    disabled={true}
    onBlur={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={-1}
    type="submit"
  >
    Add Board
  </button>
</form>
`;
